### PR TITLE
 test: makePayment_overPay (SC-4388)  

### PR DESCRIPTION
**Pull request by edag94**
# Description
testing makePayment when funds used to pay is higher than the installment

# Integrations Checklist

- [ ] Have any function signatures changed? If yes, outline below.
- [ ] Have any features changed or been added? If yes, outline below.
- [ ] Have any events changed or been added? If yes, outline below.
- [ ] Has all documentation been updated?

# Changelog

### CODE REVIEW

File: contracts/test/MapleLoanInternals.t.sol [hunk 1]
```diff
@@ -1305,10 +1305,44 @@ contract MapleLoanInternals_MakePaymentTests is TestUtils {
...
+        uint256 fundsForPaymentWithExtra = installmentToPay + amountToOverpay;
+
+        _fundsAsset.mint(address(_loan), fundsForPaymentWithExtra);
+
+        // Pay off loan with some left over.
+        ( uint256 actualPrincipal, uint256 actualInterest ) = _loan.makePayment();
+        uint256 actualInstallmentAmount = actualPrincipal + actualInterest;
+
+        assertEq(installmentToPay, actualInstallmentAmount);
+	assertEq(_loan.drawableFunds(), amountToOverpay);
```
Comment (lucas-manuel) [inline contracts/test/MapleLoanInternals.t.sol:1342]: Align params

File: contracts/test/MapleLoanInternals.t.sol [hunk 2]
```diff
@@ -1305,10 +1305,44 @@ contract MapleLoanInternals_MakePaymentTests is TestUtils {
...
-    // TODO: test_makePayment_lastPaymentClearsLoan
+        setupLoan(address(_loan), principalRequested_, paymentsRemaining_, paymentInterval_, interestRate_, endingPrincipal_);
+
+        // Drawdown all loan funds.
+        _loan.drawdownFunds(_loan.drawableFunds(), address(this));
+
+        ( uint256 expectedPrincipal, uint256 expectedInterest ) = _loan.getNextPaymentBreakdown();
 
+        uint256 installmentToPay         = expectedPrincipal + expectedInterest;
+        uint256 amountToOverpay          = 1;
```
Comment (lucas-manuel) [inline contracts/test/MapleLoanInternals.t.sol:1332]: Amount to overpay can be fuzzed since its not an edge condition

File: contracts/test/MapleLoanInternals.t.sol (hunk 1 above)
Comment (lucas-manuel) [reply contracts/test/MapleLoanInternals.t.sol:1342]: Lets also assert claimableFunds is now equal to `installmentToPay`

**Review by lucas-manuel — CHANGES_REQUESTED** [review]

**Review by lucas-manuel — APPROVED** [review]
